pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java

```diff
@@ -133,5 +142,24 @@ public ClientConfigurationData clone() {
         }
     }
 
+    public InetSocketAddress getSocks5ProxyAddress() {
+        if (Objects.nonNull(socks5ProxyAddress)) {
+            return socks5ProxyAddress;
+        }
+        String proxyAddress = System.getProperty("socks5Proxy.address");
+        try {
+            URI uri = URI.create(proxyAddress);
+            return new InetSocketAddress(uri.getHost(), uri.getPort());
+        } catch (Exception ignore) {
+            return null;
```

Comment:
Why ignore this exception?

Reply:
I will change to print a warning log instead.